apps/server/src/modules/files-storage/uc/files-storage-upload.uc.spec.ts

```diff
@@ -0,0 +1,416 @@
+			const response = createAxiosResponse(readable, headers);
+
+			return { fileRecord, userId, uploadFromUrlParams, response };
+		};
+
+		describe('WHEN user is authorised, httpService gets response and file uploads successfully', () => {
+			const setup = () => {
+				const { fileRecord, userId, uploadFromUrlParams, response } = getUploadFromUrlParams();
+
+				httpService.get.mockReturnValueOnce(of(response));
+
+				filesStorageService.uploadFile.mockResolvedValueOnce(fileRecord);
+
+				return { uploadFromUrlParams, userId, response, fileRecord };
+			};
+
+			it('should call authorizationService with correct params', async () => {
+				const { uploadFromUrlParams, userId } = setup();
+
+				await filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams);
+
+				expect(authorizationService.checkPermissionByReferences).toBeCalledWith(
+					userId,
+					uploadFromUrlParams.parentType,
+					uploadFromUrlParams.parentId,
+					{ action: Actions.write, requiredPermissions: [Permission.FILESTORAGE_CREATE] }
+				);
+			});
+
+			it('should call httpService get with correct params', async () => {
+				const { uploadFromUrlParams, userId } = setup();
+
+				await filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams);
+
+				const expectedConfig: AxiosRequestConfig = {
+					headers: uploadFromUrlParams.headers,
+					responseType: 'stream',
+				};
+
+				expect(httpService.get).toHaveBeenCalledWith(uploadFromUrlParams.url, expectedConfig);
+			});
+
+			it('should call uploadFile get with correct params', async () => {
+				const { uploadFromUrlParams, userId, response } = setup();
+
+				await filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams);
+
+				const expectedFileDescription = FileDtoBuilder.buildFromAxiosResponse(uploadFromUrlParams.fileName, response);
+				expect(filesStorageService.uploadFile).toHaveBeenCalledWith(
+					userId,
+					uploadFromUrlParams,
+					expectedFileDescription
+				);
+			});
+
+			it('should call uploadFile get with correct params', async () => {
+				const { uploadFromUrlParams, userId, fileRecord } = setup();
+
+				const result = await filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams);
+
+				expect(result).toEqual(fileRecord);
+			});
+		});
+
+		describe('WHEN user is not authorised', () => {
+			const setup = () => {
+				const { userId, uploadFromUrlParams } = getUploadFromUrlParams();
+				const error = new Error('test');
+				authorizationService.checkPermissionByReferences.mockRejectedValueOnce(error);
+
+				return { uploadFromUrlParams, userId, error };
+			};
+
+			it('should pass error', async () => {
+				const { uploadFromUrlParams, userId, error } = setup();
+
+				await expect(filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams)).rejects.toThrow(error);
+			});
+		});
+
+		describe('WHEN httpService throws error', () => {
+			const setup = () => {
+				const { userId, uploadFromUrlParams } = getUploadFromUrlParams();
+
+				const headers: AxiosResponseHeaders = {};
```

Comment:
For easyer copy paste if some other need it too in future, it make sense to put it to a named function createAxiosErrorResponse or so on top.